File: app/src/main/java/net/programmierecke/radiodroid2/station/live/metadata/lastfm/data/Track.java

package net.programmierecke.radiodroid2.station.live.metadata.lastfm.data;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Track {

    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("mbid")
    @Expose
    private String mbid;
    @SerializedName("url")
    @Expose
    private String url;
    @SerializedName("duration")
    @Expose
    private String duration;
    @SerializedName("streamable")
    @Expose
    private Streamable streamable;
    @SerializedName("listeners")
    @Expose
    private String listeners;
    @SerializedName("playcount")
    @Expose
    private String playcount;
    @SerializedName("artist")
    @Expose
    private Artist artist;
    @SerializedName("album")
    @Expose
    private Album album;
    @SerializedName("toptags")
    @Expose
    private Toptags toptags;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getMbid() {
        return mbid;
    }

    public void setMbid(String mbid) {
        this.mbid = mbid;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getDuration() {
        return duration;
    }

    public void setDuration(String duration) {
        this.duration = duration;
    }

    public Streamable getStreamable() {
        return streamable;
    }

    public void setStreamable(Streamable streamable) {
        this.streamable = streamable;
    }

    public String getListeners() {
        return listeners;
    }

    public void setListeners(String listeners) {
        this.listeners = listeners;
    }

    public String getPlaycount() {
        return playcount;
    }

    public void setPlaycount(String playcount) {
        this.playcount = playcount;
    }

    public Artist getArtist() {
        return artist;
    }

    public void setArtist(Artist artist) {
        this.artist = artist;
    }

    public Album getAlbum() {
        return album;
    }

    public void setAlbum(Album album) {
        this.album = album;
    }

    public Toptags getToptags() {
        return toptags;
    }

    public void setToptags(Toptags toptags) {
        this.toptags = toptags;
    }

}

File: app/src/main/java/net/programmierecke/radiodroid2/station/live/metadata/lastfm/data/Track.kt
package net.programmierecke.radiodroid2.station.live.metadata.lastfm.data

import com.google.gson.annotations.SerializedName

data class Track(
        @SerializedName("name")
        val name: String? = null,

        @SerializedName("mbid")
        val mbid: String? = null,

        @SerializedName("url")
        val url: String? = null,

        @SerializedName("duration")
        val duration: String? = null,

        @SerializedName("streamable")
        val streamable: Streamable? = null,

        @SerializedName("listeners")
        val listeners: String? = null,

        @SerializedName("playcount")
        val playcount: String? = null,

        @SerializedName("artist")
        val artist: Artist? = null,

        @SerializedName("album")
        val album: Album? = null,

        @SerializedName("toptags")
        val toptags: Toptags? = null
)